components/sup/src/census.rs

```diff
@@ -715,6 +754,9 @@ impl<'a> Serialize for CensusMemberProxy<'a> {
         strukt.serialize_field("suspect", &self.suspect)?;
         strukt.serialize_field("confirmed", &self.confirmed)?;
         strukt.serialize_field("departed", &self.departed)?;
+        strukt.serialize_field("service_health", &self.service_health)?;
+        strukt.serialize_field("service_health_incarnation",
```

Comment:
I do not think this is a good approach. The need to serialize the incarnation feels very wrong. I think it is possible we do not want an incarnation for service health. Instead, we should just always update the status, but inorder for this to be possible we need to be sure to garbage-collect old rumors.

Reply:
I don't think incarnation is wrong here, necessarily (it's a good and simple way to signal when information changes). Given the rumor merging logic, a new health will overwrite an old one, which is what we want. Garbage collection is (according to my perhaps fuzzy recollection) more concerned with knowing when it's OK to stop holding on to rumors that we shouldn't care about anymore; e.g., supervisor A stopped running service X 10 days ago, so the fact that we're still holding onto the fact that it was healthy back then (and potentially telling others that fact _now_) is silly.